src/utils/board/board.ts

```diff
@@ -0,0 +1,207 @@
+import {ICell} from "../../interfaces/interfaces";
+import {PieceColor, PieceState} from "../consts/piece";
+import {BOARD_SIZE} from "../consts/board";
+import canBeat from "../moves/canBeat";
+import GameStore from "../../components/store/game/GameStore";
+
+const gameStore = new GameStore();
+
+const isMinePiece = (currentPiece: ICell, selectedPiece: ICell): boolean => {
+    return currentPiece.piece?.color === selectedPiece.piece?.color;
+};
+
+const isSamePiece = (currentPiece: ICell, selectedPiece: ICell): boolean => {
+    return currentPiece.row === selectedPiece.row && currentPiece.col === selectedPiece.col;
+};
+
+const getAllowedDirections = (boardSize: number): number[] => {
+    const firstAllowedDirection = boardSize - 1;
+    const secondAllowedDirection = boardSize + 1;
+
+    return [firstAllowedDirection, secondAllowedDirection];
+};
+
+const isShouldBecomeKing = (currentPiece: ICell, selectedPiece: ICell): boolean => {
+    if (!currentPiece) {
+        return false;
+    }
+    if (selectedPiece.row === 0 && currentPiece.piece?.color === PieceColor.White) {
+        return true;
+    }
+
+    return selectedPiece.row === BOARD_SIZE - 1 && currentPiece.piece?.color === PieceColor.Black;
+};
+
+const movePiece = (
```

Comment:
I recommend you to split this functions' responsibilities a bit.

Try creating a function that just returns updated board based on your move and handle setting that to the state/setting current piece in the different place

So you'd have two functions: one that returns updated state, the second one - performs updates.
First one would become very easily testable, second would be in your component, so you wouldn't need to pass all that callbacks

Reply:
Done